data/transactions/logic/debugger.go

```diff
@@ -180,6 +200,21 @@ func (cx *evalContext) refreshDebugState() *DebugState {
 	ds.Stack = stack
 	ds.Scratch = scratch
 
+	if (cx.runModeFlags & runModeApplication) != 0 {
+		if cx.globalStateCow != nil {
+			for k, v := range cx.globalStateCow.delta {
+				ds.GlobalStateChanges[k] = v
+			}
+		}
+		for addr, cow := range cx.localStateCows {
+			delta := make(basics.StateDelta)
```

Comment:
you could preallocate this if you wanted, not that it's particularly performance sensitive...